go-controller/pkg/cni/ovs.go

```diff
@@ -98,17 +99,54 @@ func ofctlExec(args ...string) (string, error) {
 	return stdoutStr, nil
 }
 
-func waitForPodFlows(mac string) error {
+func waitForPodFlows(mac string, ipNets []*net.IPNet) error {
+	// query represents the match criteria, and different OF tables that this query may match on
+	type query struct {
+		match  string
+		tables []int
+	}
+
 	return wait.PollImmediate(200*time.Millisecond, 20*time.Second, func() (bool, error) {
-		// Query the flows by mac address
-		query := fmt.Sprintf("table=9,dl_src=%s", mac)
-		// ovs-ofctl dumps error on stderr, so stdout will only dump flow data if matches the query.
-		stdout, err := ofctlExec("dump-flows", "br-int", query)
-		if err != nil {
-			return false, nil
+		// Function checks for OpenFlow flows to know the pod is ready
+		// TODO(trozet): in the future use a more stable mechanism provided by OVN:
+		// https://bugzilla.redhat.com/show_bug.cgi?id=1839102
+
+		// Query the flows by mac address for in_port_security
+		queries := []query{
+			{
+				match:  "dl_src=" + mac,
+				tables: []int{9},
+			},
+		}
+		for _, ipNet := range ipNets {
+			var ipMatch string
+			if ipNet.IP.To4() != nil {
+				ipMatch = "ip_dst"
+			} else {
+				ipMatch = "ip6_dst"
+			}
+			// add queries for out_port_security
+			// note we need to support table 48 for 20.06 OVN backwards compatibility. Table 49 is now
+			// where out_port_security lives
```

Comment:
So if this changes again (or if that is wrong for some older release), it will completely break adding pods... So that means we are pretty strongly requiring specific very specific OVN versions... should we enforce that more strongly somewhere?

Reply:
we declared a while ago that ovn-kubernetes upstream only support latest OVN. We are kind of breaking that trend here by adding support for a previous version, but it doesn't really hurt us. In the future if OVN updates and moves these tables again we will have to add another table here, until we have a better mechanism to tell if the port is wired.

Reply:
btw the RFE for adding a better check is listed in a comment in this patch.